### PR TITLE
feat: add proxysettings for azureopenai and openai

### PR DESCRIPTION
**Pull request by tanujd11**
Closes https://github.com/k8sgpt-ai/k8sgpt-operator/issues/307

## 📑 Description

This PR adds proxyEndpoint to be configured from the operator. The feature was provided by https://github.com/k8sgpt-ai/k8sgpt/pull/987 in k8sgpt.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: pkg/resources/k8sgpt.go [hunk 1]
```diff
@@ -23,7 +23,6 @@ import (
 	"github.com/k8sgpt-ai/k8sgpt-operator/pkg/utils"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	v1 "k8s.io/api/core/v1"
```
Comment (arbreezy) [inline pkg/resources/k8sgpt.go:26]: might seem pedantic, but there is no need to remove this alias, and revert to `corev1`

**Review by arbreezy — CHANGES_REQUESTED** [review]
thank you @tanujd11 , a few comments:

- Update the docs to include the new spec's attribute
- I was checking the k8sgpt's [PR](https://github.com/k8sgpt-ai/k8sgpt/commit/307710eddc1c3f96f40a674f7dda786510e9c4cc) that we merged recently, we missed to surface the `proxyEndpoint`  as a cli [argument](https://github.com/k8sgpt-ai/k8sgpt/blob/main/cmd/auth/add.go#L147) , feel free to add it if you want
- You have to update manually the helm's chart CRD [template](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/templates/k8sgpt-crd.yaml#L19) to include the 
```
proxyEndpoint:
   type: string

File: pkg/resources/k8sgpt.go (hunk 1 above)
Comment (tanujd11) [reply pkg/resources/k8sgpt.go:26]: Hey, I just want to understand why this was imported twice here. I saw this warning on my IDE and decided to work on it. Any specific reason why is there already.

Comment (arbreezy) [reply pkg/resources/k8sgpt.go:26]: Actually, you are right, we import it twice.. ignore me, it's fine

Most likely, two different contributors added the same module, thanks for removing the duplication

**Comment by tanujd11** [comment]
> * You have to update manually the helm's chart CRD [template](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/templates/k8sgpt-crd.yaml#L19) to include the

@arbreezy I have already did the change in the file you mentioned. PTAL.

**Comment by arbreezy** [comment]
> > * You have to update manually the helm's chart CRD [template](https://github.com/k8sgpt-ai/k8sgpt-operator/blob/main/chart/operator/templates/k8sgpt-crd.yaml#L19) to include the
> 
> @arbreezy I have already did the change in the file you mentioned. PTAL.

you are right,
the only pending update is to add the new spec attribute in the [README](https://github.com/k8sgpt-ai/k8sgpt-operator/pull/348)

**Comment by arbreezy** [comment]
@tanujd11 just bumping a message, changes lgtm but please update README's example to include the new spec attribute you are creating :pray:

**Comment by tanujd11** [comment]
@arbreezy, Could you take a look? I have updated the Readme.

**Comment by tanujd11** [comment]
Can this be merged now?

**Review by arbreezy — APPROVED** [review]

**Pull request merged**